### DISCUSSION TITLE
How To Generate an Bearer Token by API Key?

### DISCUSSION THREAD

**Discussion by Stefan-WW** [skymethod/op3#22] [Q&A]
Hello, 

i couldn't find any informations how to get an Bearer Token. Can someone help me? 
i have now an api key registered.

But how to get the Bearer Token? 

As far as i know, there should be an oAuth endpoint

`curl -X POST "https://api.example.com/oauth/token" \
>      -H "Content-Type: application/json" \
>      -d '{"api_key": "YOUR_API_KEY"}'
`

But i couldn't find something like that in the op3 documentation.
The Download Button for the specification seems to be broken.
![image](https://github.com/skymethod/op3/assets/104902165/10110bdf-4e28-4c06-b1c9-d879da0bd468)

Can someone help me?

Kind Regards
Stefan

**Comment by johnspurlock-skymethod** [comment]
It's easier than that, the bearer token is available in the second box when you first create a new api key.  Think of the api key as the Account ID.  I should probably rename it.

You can regen a new bearer token for an existing api key in the https://op3.dev/api/keys page.  It's only displayed once for security purposes.

**Comment by Dwev** [comment]
For some reason, I thought the api key was the token, and I didn't realize I had to "regen" once to get my first token.
@johnspurlock-skymethod Maybe if there isn't an existing token, you could label this button "Generate token" for the first time?
